pilot/pkg/bootstrap/server.go

```diff
@@ -419,18 +421,13 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 	var clients []*mcpclient.Client
 	var conns []*grpc.ClientConn
 
-	for _, addr := range args.MCPServerAddrs {
-		u, err := url.Parse(addr)
-		if err != nil {
-			return err
-		}
-
+	for _, configSource := range s.mesh.ConfigSources {
 		securityOption := grpc.WithInsecure()
-		if u.Scheme == "mcps" {
+		if configSource.TlsSettings != nil && configSource.TlsSettings.Mode == istio_networking_v1alpha3.TLSSettings_ISTIO_MUTUAL {
```

Comment:
I am not sure if this is right. 
Should support file-based store? (this is not supported before)
Should support `TLSSettings_SIMPLE`? (Also not before)

Reply:
Yes, it should support TLS simple/mutual and also, if the address has file:/// then it should be a file based store instead of servers.